app/screens/channel_info/channel_info.js

```diff
@@ -234,6 +241,45 @@ class ChannelInfo extends PureComponent {
         toggleFavorite(currentChannel.id);
     };
 
+    handleClosePermalink = () => {
+        const {actions} = this.props;
+        actions.selectFocusedPostId('');
+        this.setState({showingPermalink: false});
+    };
+
+    handlePermalinkPress = (postId) => {
+        this.setState({isPermalink: true});
+        this.showPermalinkView(postId);
+    };
+
+    showPermalinkView = (postId) => {
+        const {actions, navigator} = this.props;
+        const {isPermalink, showingPermalink} = this.state;
```

Comment:
Same comments as above about `isPermalink` seeming unnecessary and `showingPermalink` being more reliable outside of state